fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/api/client/event/lifecycle/v1/ClientTaskScheduler.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.event.lifecycle.v1;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import net.minecraft.client.MinecraftClient;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+/**
+ * Allows the scheduling of events after a specified number of client ticks.
+ */
+@Environment(EnvType.CLIENT)
+public class ClientTaskScheduler {
+	static {
+		// Don't add this the the event handler until the class is loaded
+		ClientTickEvents.START_CLIENT_TICK.register(ClientTaskScheduler::runExecutables);
+	}
+
+	private static final List<Pair<Integer, List<ClientTickEvents.StartTick>>> executables = new ArrayList<>();
+
+	/**
+	 * Adds a new event to be run after the specified {@code delayTicks}.
+	 * @param executable the event to be run
+	 * @param delayTicks the number of ticks to wait to run the event
+	 */
+	public static void execute(ClientTickEvents.StartTick executable, int delayTicks) {
+		// See if there is already a pair with the same tick value
+		for (Pair<Integer, List<ClientTickEvents.StartTick>> pair : executables) {
+			if (pair.getLeft() == delayTicks) {
+				pair.getRight().add(executable);
+				return;
+			}
+		}
+
+		// Add the new pair
+		List<ClientTickEvents.StartTick> newList = new ArrayList<>();
+		newList.add(executable);
+		executables.add(new Pair<>(delayTicks, newList));
+	}
+
+	/**
+	 * Adds a new event to be run the next tick.
+	 * @param executable the event to be run
+	 */
+	public static void execute(ClientTickEvents.StartTick executable) {
+		execute(executable, 1);
+	}
+
+	private static void runExecutables(MinecraftClient client) {
+		for (Pair<Integer, List<ClientTickEvents.StartTick>> pair : executables) {
+			int ticksLeft = pair.getLeft() - 1;
```

Comment:
Decrementing all the items every tick seems inefficient - why not store a tickTime (tickDelta + currentTickTime at registration), then compare them at execution? Would also allow for MC's Pair to be used rather than the custom one.